### PR TITLE
Update camera samples for BUILD 2024

### PR DESCRIPTION
**Pull request by LPBourret**
- revamp Windows Studio Effect Sample:
  - replace UWP with WinUI3 and .Net6 using CSWinRT to interop with ControlMonitorHelperWinRT runtime component project
  - add support for new v2 effects
  - add button to launch Windows Settings page and Quick Settings panel
  - add background segmentation metadata visualization
  - updated readme

- modify ControlMonitorHelper C++Winrt project:
  - rename to ControlMonitorHelperWinRT
  - support KSPROPERTYSETID_ANYCAMERACONTROL
  - add better debug tracing

- modify CameraKsPropertyHelper C++WinRT project:
  - add support for Windows Studio Effects v2
  - correct FramerateThrottling SET logic where fenicky driver would fail if not sending a fully filed KSCAMERA_EXTENDEDPROP_VIDEOPROCSETTING. Retrieving the GET payload first then only changing flags or value.

- correct issues with ExtendedControlAndMetadataSampleApp

### CODE REVIEW

File: Samples/ControlMonitorApp/ControlMonitorHelper/ControlMonitorHelper.h [hunk 1]
```diff
@@ -5,12 +5,26 @@
 #pragma once
 #include "ControlMonitorManager.g.h"
 
-namespace winrt::ControlMonitorHelper::implementation
+namespace winrt::ControlMonitorHelperWinRT::implementation
 {
+    inline winrt::hstring FormatString(LPCWSTR szFormat, ...)
```
Comment (kasumman) [inline Samples/ControlMonitorApp/ControlMonitorHelper/ControlMonitorHelper.h:10]: >FormatString [](http://example.com/codeflow?start=26&length=12)

nit/tip wil has wil::str_printf that essentially does the va_args work, I don't think it directly support winrt::hstring but would support wil::unique_hstring etc

File: Samples/Shared/CameraKsPropertyHelper/KSHelper.h [hunk 2]
```diff
@@ -6,6 +6,8 @@
 
 namespace winrt::CameraKsPropertyHelper::implementation
 {
+    static GUID KSPROPERTYSETID_WindowsStudioCameraControl =
```
Comment (kasumman) [inline Samples/Shared/CameraKsPropertyHelper/KSHelper.h:9]: >KSPROPERTYSETID_WindowsStudioCameraControl [](http://example.com/codeflow?start=16&length=42)

This should be available in ks.h if newer SDK version is used. Named KSPROPERTYSETID_WindowsCameraEffect, so maybe this should use same name?

File: Samples/ControlMonitorApp/ControlMonitorHelper/ControlMonitorHelper.h (hunk 1 above)
Comment (LPBourret) [reply Samples/ControlMonitorApp/ControlMonitorHelper/ControlMonitorHelper.h:10]: cool, yeah I need to keep using hstring here for simplicity but will keep this API in mind

File: Samples/WindowsStudio/WindowsStudioSample_WinUI/KsHelper.cs [hunk 3]
```diff
@@ -238,5 +347,40 @@ public static KsBackgroundCameraExtendedPropPayload FromBytes(byte[] bytes, int
...
+        {
+            SoftwareBitmap maskSoftwareBitmap = null;
+            if (frame.Properties.TryGetValue(MFSampleExtension_CaptureMetadata, out var captureMetadata))
+            {
+                if (captureMetadata is IReadOnlyDictionary<Guid, object> captureMetadataLookUp)
+                {
+                    if (captureMetadataLookUp.TryGetValue(MF_CAPTURE_METADATA_FRAME_BACKGROUND_MASK, out var backgroundSegmentationMask))
+                    {
+                        byte[] payloadBytes = (byte[])backgroundSegmentationMask;
+                        //KSCAMERA_METADATA_BACKGROUNDSEGMENTATIONMASK payload = Marshal.PtrToStructure<KSCAMERA_METADATA_BACKGROUNDSEGMENTATIONMASK>((IntPtr)backgroundSegmentationMask);
```
Comment (kasumman) [inline Samples/WindowsStudio/WindowsStudioSample_WinUI/KsHelper.cs:361]: remove the commented code lines?

**Pull request merged**